crates/admin/src/cluster_controller/service.rs

```diff
@@ -1227,61 +1133,40 @@ mod tests {
         let interval_duration = Duration::from_secs(10);
         let admin_options = AdminOptionsBuilder::default()
             .log_trim_check_interval(interval_duration.into())
-            .build()
-            .unwrap();
+            .build()?;
         let mut config: Configuration = Configuration {
             networking,
             admin: admin_options,
             ..Default::default()
         };
         config.bifrost.default_provider = ProviderKind::InMemory;
+        // this configuration will activate archived LSN-only trimming, even for single nodes
         config.worker.snapshots.destination = Some("a-repository-somewhere".to_string());
 
-        let lsn = AtomicU64::new(0);
-        let applied_persisted_lsn = Arc::new(lsn);
+        let applied_lsn = Arc::new(AtomicU64::new(0));
+        let persisted_lsn = Arc::new(AtomicU64::new(0));
+        let archived_lsn = Arc::new(AtomicU64::new(0));
 
-        let (node_env, bifrost, cluster_state) = create_test_env(config, |builder| {
-            let get_node_state_handler = MockNodeStateHandler {
-                applied_lsn: applied_persisted_lsn.clone(),
-                persisted_lsn: applied_persisted_lsn.clone(),
-                archived_lsn: Arc::new(AtomicU64::new(0)), // unused in this test
-                block_list: Default::default(),
+        let (_, bifrost, cluster_state) = create_test_env_with_nodes(config, 1, |builder| {
+            let n1_msg_handler = MockNodeStateHandler {
+                applied_lsn: applied_lsn.clone(),
+                persisted_lsn: persisted_lsn.clone(),
+                archived_lsn: archived_lsn.clone(),
             };
-
             builder
-                .add_message_handler(get_node_state_handler)
+                .add_message_handler(n1_msg_handler)
                 .add_message_handler(NoOpMessageHandler::<ControlProcessors>::default())
         })
         .await?;
 
-        let get_node_2_state_handler = MockNodeStateHandler {
-            applied_lsn: applied_persisted_lsn.clone(),
-            persisted_lsn: applied_persisted_lsn.clone(),
-            archived_lsn: Arc::new(AtomicU64::new(0)), // unused in this test
-            block_list: Default::default(),
-        };
-        let node_2 = MockPeerConnection::connect(
-            GenerationalNodeId::new(2, 2),
-            node_env.metadata.nodes_config_version(),
-            node_env
-                .metadata
-                .nodes_config_ref()
-                .cluster_name()
-                .to_owned(),
-            node_env.networking.connection_manager(),
-            10,
-        )
-        .await?;
-        // let node2 receive messages and use the same message handler as node1
-        let (_node_2, _node2_reactor) =
-            node_2.process_with_message_handler(get_node_2_state_handler)?;
+        // N2 does not react to messages and will be flagged as dead
```

Comment:
Where does N2 come from? The test name reads as if there is only a single node.

Reply:
My mistake; this comment is for `cluster_auto_trim_with_dead_nodes_present()` below - moved!